modules/standard/AutoMath.chpl

```diff
@@ -217,9 +222,20 @@ module AutoMath {
     return fabsf(x);
   }
 
+  /* Return the absolute value of a param real(32) as a param */
+  proc abs(param x: real(32)) param : real(32) {
+    return if x < 0 then -x else x;
+  }
```

Comment:
Floating points are weird, and this worries me a little. Consider the following Python:

```Python
>>> float("-0")
-0.0
>>> float("-0") < 0
False
>>> abs(float("-0"))
0.0
```

In other words, numerically this differs from the behavior of Python's `abs`:

```Python
>>> def myabs(x): return -x if x < 0 else x
...
>>> myabs(float("-0"))
-0.0
>>> abs(float("-0"))
0.0
```

Reply:
I can wire this up to call an `abs` primitive in the compiler to run `fabs`. Do you feel that would address the issue?

Reply:
Yes, I do -- `fabs` seems ideal.